tests/e2e/snapshots.go

```diff
@@ -0,0 +1,120 @@
+package e2e
+
+import (
+	"context"
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epv "k8s.io/kubernetes/test/e2e/framework/pv"
+	"k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var APIGroup = "snapshot.storage.k8s.io"
+
+type snapshotsTestSuite struct {
+	tsInfo storageframework.TestSuiteInfo
+}
+
+// InitCustomSnapshotsTestSuite returns snapshotsTestSuite that implements TestSuite interface
+// using custom test patterns
+func InitCustomSnapshotsTestSuite(patterns []storageframework.TestPattern) storageframework.TestSuite {
+	return &snapshotsTestSuite{
+		tsInfo: storageframework.TestSuiteInfo{
+			Name:         "snapshots",
+			TestPatterns: patterns,
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: "1Mi",
+			},
+			FeatureTag: "[Feature:VolumeSnapshotDataSource]",
+		},
+	}
+}
+
+// InitSnapshotsTestSuite returns snapshotsTestSuite that implements TestSuite interface\
+// using test suite default patterns
+func InitSnapshotsTestSuite() storageframework.TestSuite {
+	patterns := []storageframework.TestPattern{
+		storageframework.DynamicSnapshotDelete,
+		storageframework.DynamicSnapshotRetain,
+		storageframework.PreprovisionedSnapshotDelete,
+		storageframework.PreprovisionedSnapshotRetain,
+	}
+	return InitCustomSnapshotsTestSuite(patterns)
+}
+
+func (s *snapshotsTestSuite) GetTestSuiteInfo() storageframework.TestSuiteInfo {
+	return s.tsInfo
+}
+
+func (s *snapshotsTestSuite) SkipUnsupportedTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+	_, ok := driver.(FSxTestDriver)
+	if !ok {
+		skipper.Skipf("Driver %q does not support dynamic provisioning - skipping", driver.GetDriverInfo().Name)
```

Comment:
Why do we skip this (and can you add a comment explaining why)? Otherwise this is misleading, because the driver supports dynamic provisioning.

Reply:
Good catch, I forgot to change the skip message. Pretty much this skips the testsuite if the driver supplied is not an FSx for OpenZFS driver. This is because the testsuite is specifically designed for our driver.

The testsuites should be thought of as an individual component. It can't be assumed that only our driver will use the testsuite.